### PR TITLE
Initial version of cticker

### PR DESCRIPTION
**Pull request by stevenh**
Initial version of cticker which is a library that provides a ticker which is reliable under clock drift and clock adjustments, that is if you ask it to tick on the minute it will ensure that it does so even if the underlying clock is inaccurate or gets adjusted.

### CODE REVIEW

File: ticker.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	}
+}
+
+// sends ticks to t.C with the tickers defined accuracy.
+func (t *Ticker) tick(c chan time.Time) {
+	ticks := t.ticker.Ch()
+	for {
+		select {
+		case now := <-ticks:
+			fmt.Println("tick:", now)
```
Comment (milonoir) [inline ticker.go]: Are Println() calls intentional? (more below)

File: ticker_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	period time.Duration
+	times  []time.Time
+}
+
+func (t *mockTicker) run() {
+	tick := time.NewTicker(t.period)
+	defer tick.Stop()
+
+	for _, tm := range t.times {
+		fmt.Println("run:", tm)
```
Comment (milonoir) [inline ticker_test.go]: Are Println() calls intentional? (more below)

Comment (stevenh) [reply ticker_test.go]: Yes absolutely, I left them in temporarily so it gives visibility when you download and test.

We will remove these before merge.

File: ticker.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	c := make(chan time.Time, 1)
+	t := &Ticker{
+		C:    c,
+		d:    d,
+		a:    accuracy,
+		done: make(chan struct{}),
+	}
+
+	go func() {
+		// Syncronise to the accuracy.
```
Comment (milonoir) [inline ticker.go]: typo: synchronise

File: ticker.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+			t.ticker = newTicker(accuracy)
+			go t.tick(c)
+		}
+
+	}()
+
+	return t
+}
+
+// Stop turns off a ticker. After Stop, no more ticks will be sent.
```
Comment (milonoir) [inline ticker.go]: "Stop turns off the ticker."

File: ticker.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+
+	close(t.done)
+	if t.ticker != nil {
+		t.ticker.Stop()
+	}
+}
+
+// sends ticks to t.C with the tickers defined accuracy.
```
Comment (milonoir) [inline ticker.go]: missing method name: "tick sends ticks to ..."

**Review by milonoir — CHANGES_REQUESTED** [review]

**Review by milonoir — APPROVED** [review]

**Review by milonoir — APPROVED** [review]

**Pull request merged**